From 6f4e7ae07d65574c3505f40bf7b06db641876ac4 Mon Sep 17 00:00:00 2001
From: Topi Reinio <topi.reinio@qt.io>
Date: Thu, 13 Aug 2020 12:35:36 +0200
Subject: [PATCH 078/171] Doc: Fix documentation warnings for Qt XML

Content for the deprecated SAX interface was removed, remove rest of
the linking as well.

Change-Id: Idd233a821139837b779e551aeba914114500e39e
Reviewed-by: Paul Wicking <paul.wicking@qt.io>
(cherry picked from commit 0ac7a81250e7291d12d9a176197260344884419c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/xml/sax/qxml.cpp | 40 ++++++++++++----------------------------
 1 file changed, 12 insertions(+), 28 deletions(-)

diff --git a/src/xml/sax/qxml.cpp b/src/xml/sax/qxml.cpp
index ea6124f45d..7a444713c1 100644
--- a/src/xml/sax/qxml.cpp
+++ b/src/xml/sax/qxml.cpp
@@ -523,8 +523,6 @@ public:
     pushContext() to start a new namespace context, and popContext()
     to return to the previous namespace context. Use splitName() or
     processName() to split a name into its prefix and local name.
-
-    \sa {Namespace Support via Features}
 */
 
 /*!
@@ -807,8 +805,6 @@ QXmlAttributes::~QXmlAttributes()
     Looks up the index of an attribute by the qualified name \a qName.
 
     Returns the index of the attribute or -1 if it wasn't found.
-
-    \sa {Namespace Support via Features}
 */
 int QXmlAttributes::index(const QString& qName) const
 {
@@ -840,8 +836,6 @@ int QXmlAttributes::index(QLatin1String qName) const
     name.
 
     Returns the index of the attribute, or -1 if it wasn't found.
-
-    \sa {Namespace Support via Features}
 */
 int QXmlAttributes::index(const QString& uri, const QString& localPart) const
 {
@@ -874,8 +868,6 @@ int QXmlAttributes::length() const
     Looks up an attribute's local name for the attribute at position
     \a index. If no namespace processing is done, the local name is
     an empty string.
-
-    \sa {Namespace Support via Features}
 */
 QString QXmlAttributes::localName(int index) const
 {
@@ -885,8 +877,6 @@ QString QXmlAttributes::localName(int index) const
 /*!
     Looks up an attribute's XML 1.0 qualified name for the attribute
     at position \a index.
-
-    \sa {Namespace Support via Features}
 */
 QString QXmlAttributes::qName(int index) const
 {
@@ -897,8 +887,6 @@ QString QXmlAttributes::qName(int index) const
     Looks up an attribute's namespace URI for the attribute at
     position \a index. If no namespace processing is done or if the
     attribute has no namespace, the namespace URI is an empty string.
-
-    \sa {Namespace Support via Features}
 */
 QString QXmlAttributes::uri(int index) const
 {
@@ -959,8 +947,6 @@ QString QXmlAttributes::value(int index) const
 
     Returns an attribute's value for the qualified name \a qName, or an
     empty string if no attribute exists for the name given.
-
-    \sa {Namespace Support via Features}
 */
 QString QXmlAttributes::value(const QString& qName) const
 {
@@ -975,8 +961,6 @@ QString QXmlAttributes::value(const QString& qName) const
 
     Returns an attribute's value for the qualified name \a qName, or an
     empty string if no attribute exists for the name given.
-
-    \sa {Namespace Support via Features}
 */
 QString QXmlAttributes::value(QLatin1String qName) const
 {
@@ -1497,7 +1481,7 @@ QString QXmlInputSource::fromRawData(const QByteArray &data, bool beginning)
     startPrefixMapping() is called.
 
     \sa QXmlDTDHandler, QXmlDeclHandler, QXmlEntityResolver, QXmlErrorHandler,
-        QXmlLexicalHandler, {Introduction to SAX2}
+        QXmlLexicalHandler
 */
 
 /*!
@@ -1571,7 +1555,7 @@ QString QXmlInputSource::fromRawData(const QByteArray &data, bool beginning)
     reports an error. The reader uses the function errorString() to
     get the error message.
 
-    \sa endPrefixMapping(), {Namespace Support via Features}
+    \sa endPrefixMapping()
 */
 
 /*!
@@ -1584,7 +1568,7 @@ QString QXmlInputSource::fromRawData(const QByteArray &data, bool beginning)
     reports an error. The reader uses the function errorString() to
     get the error message.
 
-    \sa startPrefixMapping(), {Namespace Support via Features}
+    \sa startPrefixMapping()
 */
 
 /*!
@@ -1616,7 +1600,7 @@ QString QXmlInputSource::fromRawData(const QByteArray &data, bool beginning)
     reports an error. The reader uses the function errorString() to
     get the error message.
 
-    \sa endElement(), {Namespace Support via Features}
+    \sa endElement()
 */
 
 /*!
@@ -1630,7 +1614,7 @@ QString QXmlInputSource::fromRawData(const QByteArray &data, bool beginning)
     reports an error. The reader uses the function errorString() to
     get the error message.
 
-    \sa startElement(), {Namespace Support via Features}
+    \sa startElement()
 */
 
 /*!
@@ -1720,7 +1704,7 @@ QString QXmlInputSource::fromRawData(const QByteArray &data, bool beginning)
     with the error text being reported with errorString().
 
     \sa QXmlDTDHandler, QXmlDeclHandler, QXmlContentHandler, QXmlEntityResolver,
-        QXmlLexicalHandler, {Introduction to SAX2}
+        QXmlLexicalHandler
 */
 
 /*!
@@ -1797,7 +1781,7 @@ events are reported.
     unparsedEntityDecl() respectively.
 
     \sa QXmlDeclHandler, QXmlContentHandler, QXmlEntityResolver, QXmlErrorHandler,
-        QXmlLexicalHandler, {Introduction to SAX2}
+        QXmlLexicalHandler
 */
 
 /*!
@@ -1861,7 +1845,7 @@ events are reported.
     QXmlReader::setEntityResolver().
 
     \sa QXmlDTDHandler, QXmlDeclHandler, QXmlContentHandler, QXmlErrorHandler,
-        QXmlLexicalHandler, {Introduction to SAX2}
+        QXmlLexicalHandler
 */
 
 /*!
@@ -1927,7 +1911,7 @@ events are reported.
     endEntity(), startCDATA(), endCDATA() and comment() functions.
 
     \sa QXmlDTDHandler, QXmlDeclHandler, QXmlContentHandler, QXmlEntityResolver,
-        QXmlErrorHandler, {Introduction to SAX2}
+        QXmlErrorHandler
 */
 
 /*!
@@ -2073,7 +2057,7 @@ events are reported.
     externalEntityDecl() functions.
 
     \sa QXmlDTDHandler, QXmlContentHandler, QXmlEntityResolver, QXmlErrorHandler,
-        QXmlLexicalHandler, {Introduction to SAX2}
+        QXmlLexicalHandler
 */
 
 /*!
@@ -2201,7 +2185,7 @@ events are reported.
     document has been read completely.
 
     \sa QXmlDTDHandler, QXmlDeclHandler, QXmlContentHandler, QXmlEntityResolver,
-        QXmlErrorHandler, QXmlLexicalHandler, {Introduction to SAX2}
+        QXmlErrorHandler, QXmlLexicalHandler
 */
 
 /*!
@@ -3012,7 +2996,7 @@ bool QXmlSimpleReader::feature(const QString& name, bool *ok) const
             data in one chunk.
     \endtable
 
-    \sa feature(), hasFeature(), {SAX2 Features}
+    \sa feature(), hasFeature()
 */
 void QXmlSimpleReader::setFeature(const QString& name, bool enable)
 {
-- 
2.31.1

